src/cluster.c

```diff
@@ -4538,7 +4636,7 @@ void clusterCron(void) {
         myself->slaveof &&
         nodeHasAddr(myself->slaveof))
     {
-        replicationSetMaster(myself->slaveof->ip, myself->slaveof->port);
+        replicationSetMaster(myself->slaveof->ip, getNodeDefaultClientPort(myself->slaveof));
```

Comment:
Isn't this wrong? if `tls-replication` is set to true, we should try to connect with the the tls_port, and if it is set to true we should connect to the tcp port.